tests/Affects/Mails/Unit/ConfigureMailsTest.php

```diff
@@ -16,13 +16,49 @@
 
 namespace Tenancy\Tests\Affects\Mails\Unit;
 
+use Illuminate\Mail\Transport\ArrayTransport;
+use Mockery;
 use Tenancy\Affects\Mails\Events\ConfigureMails;
 use Tenancy\Affects\Mails\Provider;
+use Tenancy\Facades\Tenancy;
 use Tenancy\Tests\Affects\AffectsEventUnitTestCase;
 
 class ConfigureMailsTest extends AffectsEventUnitTestCase
 {
     protected $affectsProvider = Provider::class;
 
     protected $event = ConfigureMails::class;
+
+    /** @test */
+    public function the_event_can_replace_swiftmailer()
+    {
+        $this->app->register($this->affectsProvider);
+
+        $this->events->listen($this->event, function (ConfigureMails $event) {
+            $event->replaceSwiftMailer(new ArrayTransport());
+
+            $this->assertInstanceOf(ArrayTransport::class, $event->mailer->getSwiftMailer()->getTransport());
```

Comment:
I would prefer moving this under the `setTenant`. Right now it feels like a weird assertion 😅 
Yes, it is a test, but at the same time, we want to know if it's the ArrayTransport when the Developer uses the Mailer :)